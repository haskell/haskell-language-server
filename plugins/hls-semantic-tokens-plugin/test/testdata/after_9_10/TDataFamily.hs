{-# LANGUAGE TypeFamilies #-}
module TDatafamily where

-- Declare a list-like data family
data family XList a

-- Declare a list-like instance for Char
data instance XList Char = XCons !Char !(XList Char) | XNil

-- Declare a number-like instance for ()
data instance XList () = XListUnit !Int
