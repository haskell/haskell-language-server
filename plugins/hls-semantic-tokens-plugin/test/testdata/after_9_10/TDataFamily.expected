2:8-19 TModule "TDatafamily"
5:13-18 TTypeFamily "XList"
5:19-20 TTypeVariable "a"
8:15-20 TTypeFamily "XList"
8:21-25 TTypeConstructor "Char"
8:28-33 TDataConstructor "XCons"
8:35-39 TTypeConstructor "Char"
8:42-47 TTypeFamily "XList"
8:48-52 TTypeConstructor "Char"
8:56-60 TDataConstructor "XNil"
11:15-20 TTypeFamily "XList"
11:26-35 TDataConstructor "XListUnit"
11:37-40 TTypeConstructor "Int"
